frontend/packages/ux-editor/src/components/Properties/PropertiesHeader/EditLayoutSetForSubform/EditLayoutSet/CreateNewSubformSection/CreateNewSubformSection.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import {
+  StudioCard,
+  StudioParagraph,
+  StudioRecommendedNextAction,
+  StudioTextfield,
+} from '@studio/components';
+import { useValidateLayoutSetName } from 'app-shared/hooks/useValidateLayoutSetName';
+import type { LayoutSets } from 'app-shared/types/api/LayoutSetsResponse';
+import { SubformDataModel } from './SubformDataModel';
+import { CreateNewSubformButtons } from './CreateNewSubformButtons';
+import { SubformInstructions } from './SubformInstructions';
+import { useCreateSubform } from '@altinn/ux-editor/hooks/useCreateSubform';
+
+type CreateNewSubformSectionProps = {
+  layoutSets: LayoutSets;
+  setShowCreateSubformCard: (showCreateSubform: boolean) => void;
+  hasSubforms: boolean;
+  recommendedNextActionText: {
+    title: string;
+    description: string;
+  };
+  onComponentUpdate: (subform: string) => void;
+};
+
+export const CreateNewSubformSection = ({
+  layoutSets,
+  setShowCreateSubformCard,
+  hasSubforms,
+  recommendedNextActionText: { title, description },
+  onComponentUpdate,
+}: CreateNewSubformSectionProps): React.ReactElement => {
+  const { t } = useTranslation();
+  const { validateLayoutSetName } = useValidateLayoutSetName();
+  const [nameError, setNameError] = useState<string | Boolean>(true);
+  const [dataModel, setDataModel] = useState<string>('');
+  const [displayDataModelInput, setDisplayDataModelInput] = useState(false);
+  const { createSubform, isPendingNewSubformMutation } = useCreateSubform();
+
+  const handleSubformName = (subformName: string) => {
+    const subformNameValidation = validateLayoutSetName(subformName, layoutSets);
+    setNameError(subformNameValidation);
+  };
+
+  const handleDataModel = (dataModelId: string) => {
+    // Add data model validation here and replace dataModel useState with dataModelError
+    // when this PR is merged: https://github.com/Altinn/altinn-studio/issues/13364
```

Comment:
I’d suggest creating a new subtask under #13764 for this task
```suggestion
    // TODO: https://github.com/Altinn/altinn-studio/issues/xxxxx
```

Reply:
I agree, here is the subtask https://github.com/Altinn/altinn-studio/issues/14184, and I will change the comment :)